util/reggen/field.py

```diff
@@ -118,6 +142,20 @@ def from_raw(reg_name: str,
         bits = Bits.from_raw(where, reg_width, params, rd['bits'])
 
         raw_resval = rd.get('resval')
+
+        if raw_mubi:
+            # When mubi type, the resval supplied is a boolean which is converted
+            # to a mubi value
+            chk_resval = check_bool(raw_resval, 'resval field for {}'.format(where))
+
+            # convert value to a string
+            mubi_type = f'mubi{bits.width()}'
+            if mubi_type not in MUBI_TYPES:
+                raise ValueError(f'mubi field for {name} does not support width '
+                                 f'of {bits.width()}')
+
+            raw_resval = Field.get_mubi_val(mubi_type, chk_resval)
```

Comment:
If we go with my proposed refactor above, this would be something like:
```python
            # convert value to a string
            if bits.width() not in MUBI_RAW_VALS:
                raise ValueError(f'mubi field for {name} has unsupported '
                                 f'width: {bits.width()}')

            raw_resval = Field.get_mubi_val(bits.width(), chk_resval)
```
(Note that it's fine for `raw_resval` to be an integer: `check_xint` supports that, and it seems a bit silly to print and then re-parse a string).